### DISCUSSION TITLE
Help with oven configuration required

### DISCUSSION THREAD

**Discussion by sarfaraz22** [thoukydides/homebridge-homeconnect#96] [Q&A] [answered]
Hi, 
First of, many thanks for creating this plugin. I have installed the plugin to control 2 of my home-connect enabled Siemen ovens.
I have 2 questions:-
1. I can start various programs like 4D air etc. It defaults to 1 min. I am not seeing any way to set the temperature and duration.
2.  I get a warning in my HB log : `Claims can be both switched off and placed in standby; treating as cannot be switched off`. This stops me from turning the oven on or setting it back to standby. The oven do not support turning it off. Is there a workaround to resolve this issue ?

**Comment by thoukydides** [comment, marked as answer]
> I can start various programs like 4D air etc. It defaults to 1 min. I am not seeing any way to set the temperature and duration.

Please read the documentation:
* https://github.com/thoukydides/homebridge-homeconnect/wiki/config.json
* https://github.com/thoukydides/homebridge-homeconnect/wiki/Programs

Unfortunately, the configuration for programs is a bit complicated. It is far easier if you use the [homebridge-config-ui-x](https://github.com/oznu/homebridge-config-ui-x) web interface.

> I get a warning in my HB log : Claims can be both switched off and placed in standby; treating as cannot be switched off. This stops me from turning the oven on or setting it back to standby. The oven do not support turning it off. Is there a workaround to resolve this issue ?

This is a Home Connect API bug.

There is a lot of variety in how different appliances support the Home Connect API, so this plugin attempts to determine from the API the supported features of each appliance. In this case the API is currently returning incorrect (and contradictory) information for some appliances, so the plugin falls back on the safe option of not enabling the functionality.

I have reported this issue to the Home Connect team and they have promised a fix. (Actually, this same issue has occurred and been fixed previously - which is why the plugin already implements the workaround of disabling power control when conflicting capabilities are erroneously reported.)
